py_ecc/fields/optimized_field_elements.py

```diff
@@ -421,6 +422,23 @@ def __init__(self, coeffs: Sequence[IntOrFQ]) -> None:
         self.mc_tuples = [(i, c) for i, c in enumerate(self.FQ2_MODULUS_COEFFS) if c]
         super().__init__(coeffs, self.FQ2_MODULUS_COEFFS)
 
+    @property
+    def sgn0(self: T_FQP) -> int:
+        """
+        Calculates the sign of a value.
+        sgn0(x) = 1 when x is 'negative'; otherwise, sg0(x) = 0
+
+        Note this is an optimized variant for m = 2
+
+        Defined here:
+        https://tools.ietf.org/html/draft-irtf-cfrg-hash-to-curve-07#section-4.1
+        """
+        x_0, x_1 = self.coeffs
+        sign_0 = x_0 % 2
```

Comment:
The modulus coefficients can be `FQ` elements or `int`s, so I think `mod_2(x)` should be implemented as a function with separate cases for int and FQ.

Reply:
👍    Added `mod_int` helper.